var/spack/repos/builtin/packages/emacs/package.py

```diff
@@ -60,6 +60,14 @@ def configure_args(self):
                 '--with-x-toolkit={0}'.format(toolkit)
             ]
         else:
-            args = ['--without-x']
+            args = ['--without-x',
+                    '--with-jpeg=no',
+                    '--with-gif=no',
+                    '--with-tiff=no',
+                    '--with-xpm=no',
+                    '--with-x-toolkit=no',
+                    ]
+            if '+gtkplus' in spec:
+                tty.error('The variant +gtkplus is ignored if ~X is selected.')
```

Comment:
sorry, i was not precise enough. I was thinking `raise InstallError()` so that `~x+gtkplus` is not possible.